libbeat/common/field.go

```diff
@@ -292,6 +292,46 @@ func (f Fields) getKeys(namespace string) []string {
 	return keys
 }
 
+// canConcat checks if inside fields the given node can be concatenated.
+// If the node already exists, this is only allowed for nodes of type object.
+func (f Fields) CanConcat(key string) bool {
```

Comment:
Can we clarify the comment for when fields are concatenable. How about naming it `conflicts` ? This is what the for loop on `ConcatFields` does, look for conflicts between 2 field arrays.

`canConcat` checks that no node is reachable by key AND that the common prefix between all fields and the path given by key resolves into an object-node.

We should return an error describing why `key` conflicts with `f`. Moving the loop from ConcatFields into a separate loop allows us to cleanly collect N errors into a multierror.

Do we need to export `CanConcat`? We seem to export too many symbols already. Let's try to not export symbols unless we know we will need them.